IdentityCore/src/workplacejoin/mac/MSIDWorkPlaceJoinUtil.m

```diff
@@ -28,11 +28,44 @@
 #import "MSIDWorkplaceJoinChallenge.h"
 #import "MSIDWPJKeyPairWithCert.h"
 #import "MSIDWorkPlaceJoinUtilBase+Internal.h"
+#import "MSIDExternalSSOContext.h"
+#import "MSIDAADAuthority.h"
 
 // Convenience macro to release CF objects
 
 @implementation MSIDWorkPlaceJoinUtil
 
++ (MSIDWPJKeyPairWithCert *)wpjKeyPairWithSSOContext:(MSIDExternalSSOContext *)ssoContext
+                                            tenantId:(NSString *)tenantId
+                                             context:(id<MSIDRequestContext>)context
+{
+    if (![NSString msidIsStringNilOrBlank:tenantId])
+    {
+        NSURL *tokenEndpointURL = ssoContext.tokenEndpointURL;
+        
+        if (tokenEndpointURL)
+        {
+            NSError *authorityURLError = nil;
+            MSIDAADAuthority *authorityURL = [[MSIDAADAuthority alloc] initWithURL:tokenEndpointURL
+                                                                         rawTenant:nil
+                                                                           context:context
+                                                                             error:&authorityURLError];
+            
+            if (!authorityURL)
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Failed to create authority URL with error %@", authorityURLError);
```

Comment:
Should return here?

Reply:
+1

Reply:
I thought it's better to not return here because we don't know if tenant is matching or not, since we failed to create the authority

Reply:
Added return for now, will test a bit more and make a change to remove it if needed later